tests/e2e/vmc_csi_deployments.go

```diff
@@ -0,0 +1,196 @@
+		deployment.Spec.Strategy.RollingUpdate.MaxSurge = &intstr.IntOrString{Type: intstr.Int, IntVal: int32(0)}
+		deployment.Spec.Strategy.RollingUpdate.MaxUnavailable = &intstr.IntOrString{Type: intstr.Int, IntVal: int32(1)}
+		deployment, err = client.AppsV1().Deployments(namespace).Update(ctx, deployment, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = fdep.WaitForDeploymentComplete(client, deployment)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pods, err := fdep.GetPodsForDeployment(client, deployment)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pod := pods.Items[0]
+		err = fpod.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Update deployment image")
+		deployment, err = client.AppsV1().Deployments(namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		deployment.Spec.Template.Spec.Containers[0].Image = "k8s.gcr.io/nginx-slim:0.9"
+		deployment, err = fdep.UpdateDeploymentWithRetries(client, namespace, deployment.Name, deployment.DeepCopyInto)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		err = fdep.WaitForDeploymentComplete(client, deployment)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Scale down replicas")
+		updateDeploymentReplica(client, replica-1, deployment.Name, namespace)
+
+		deployment, err = client.AppsV1().Deployments(namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("replica count after scale down of deployments %v", deployment.Status.Replicas)
+		gomega.Expect((deployment.Status.Replicas) == replica-1).To(gomega.BeTrue(),
+			"Number of containers in the deployment should match with number of replicas")
+		err = fpod.WaitForPodNotFoundInNamespace(client, pod.Name, namespace, pollTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Scale up replicas")
+		updateDeploymentReplica(client, replica, deployment.Name, namespace)
+		deployment, err = client.AppsV1().Deployments(namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("replica count after scale up of deployments %v", deployment.Status.Replicas)
+		gomega.Expect((deployment.Status.Replicas) == replica).To(gomega.BeTrue(),
+			"Number of containers in the deployment should match with number of replicas")
+
+		pods, err = fdep.GetPodsForDeployment(client, deployment)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pod = pods.Items[0]
+		err = fpod.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Delete Deployment")
+		err = client.AppsV1().Deployments(namespace).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		deployment = nil
+
```

Comment:
we should check that pod dies after we delete the deployment

Reply:
Added a check for the same